### ISSUE TITLE
emacsWithPackagesFromUsePackage stopped working after last update

### ISSUE THREAD

**Issue by 0x450x6c** [nix-community/emacs-overlay#45]
```
error: value is a string while a set was expected, at /nix/store/5nwss7arwhryix0b5ijm0dhlq09jal6k-source/repos/fromElisp/default.nix:485:19
(use '--show-trace' to show detailed location information)
```

https://github.com/nix-community/emacs-overlay/blob/fd1e83d00504ff04245f2d61f40fdbede700a261/repos/fromElisp/default.nix#L485

**Comment by adisbladis** [comment]
@0x450x6c Could you provide some more context? It'd be great if you could pastebin your configuration.

**Comment by 0x450x6c** [comment]
@adisbladis 
There is error with following emacs configuration:
```
(setq user-emacs-directory `,(expand-file-name user-emacs-directory))
```

But no errors when using following:
```
(setq user-emacs-directory (expand-file-name user-emacs-directory))
```

**Comment by adisbladis** [comment]
cc @talyz

**Comment by cmm** [comment]
Please excuse the idle question (I'm too new to Nix yet to contribute more meaningfully), but cannot Emacs itself be used to parse (i.e. `read`) user's .emacs file?  There must be is a point in the derivation process where Emacs (sans additional packages, however those are derived) is already available, right?  Then getting the list of "ensured" packages would be just a matter of doing something like `${emacs}/bin/emacs --batch -q --eval 'load ${src}/derive-ensured-pkgs.el' ${dotEmacs}`, and `derive-ensured-pkgs.el` being something like:
```
(with-temp-buffer
  (insert-file-contents-literally (car (last command-line-args)))
  (while (not (eobp))
    (let ((form (read)))
      (when (eq (car form) 'use-package)
        # you get the idea...
```

In short, you'd just use the native Emacs Lisp reader in a pure manner?

**Comment by adisbladis** [comment]
@cmm You could do that, but with some nasty drawbacks, namely IFD (import-from-derivation).

I'm quoting what the nix irc bot has to say about IFD:
> import-from-derivation (IFD) is when you evaluate nix from a derivation result, for example `import (pkgs.writeText "n" "1 + 1")` will evaluate to 2. This is sometimes problematic because it requires evaluating some, building some, and then evaluating the build result. It has been described as "such a nice footgun."

IFD has more drawbacks than what's being described here.

**Comment by talyz** [comment]
I just pulled in an update to `fromElisp` which fixes this. The quote parsing couldn't handle stacked quotes and had to be rewritten.

**Comment by cmm** [comment]
@adisbladis [ at the risk of veering far into counterfactual theoretical windbaggery land -- feel free to ignore me ]

if a world where this IFD-vs-Nix-elisp-reader dichotomy is false is not a refactoring away, then either I don't understand something very basic about Nix (which is entirely possible), or this is just... sad.

**Comment by talyz** [comment]
@cmm If you want to use emacs to do the same thing, me and @xantoz cooked up a solution for exactly that a while ago at https://github.com/talyz/nix-emacs-with-use-package-pkgs. I was using that, before writing this, without any major issues, but it does, as @adisbladis mentioned, use IFD with its potential drawbacks.
